### PR TITLE
docs: update framework integrations docs

### PR DESCRIPTION
**Pull request by devcorpio**
# Summary

The goal of this documentation update is to highlight two things:

- RUM framework specific integration is a recommendation and not a requirement
- Enumerate the reasons why we recommend framework specific integration (Vue, React, Angular)

@vigneshshanmugam as we discussed a few days ago, the changes are added to the [frameworks](https://www.elastic.co/guide/en/apm/agent/rum-js/current/supported-technologies.html#frameworks) section of the [supported technologies page](https://www.elastic.co/guide/en/apm/agent/rum-js/current/supported-technologies.html)

Screenshot of the preview (edit: includes Paul's suggestions):
Screenshot 2023-04-05 at 14 51 18

### CODE REVIEW

File: docs/supported-technologies.asciidoc [hunk 1]
```diff
@@ -117,6 +117,12 @@ The agent supports <<framework-integrations,integrations with certain frameworks
 To instrument custom metrics, like rendering time or mounting time of components on frameworks like React, Angular, Vue,
 etc., use the <<custom-transactions,custom transactions API>>.
 
+The core RUM agent supports MPAs and SPAs out of the box. This means that for SPAs to use <<framework-integrations,framework specific integrations>> is a recommendation rather than a requirement. We recommend them for the following reasons:
```
Comment (paulb-elastic) [inline docs/supported-technologies.asciidoc:120]: Might be overkill and obvious, but maybe add some detail for clarity, for example:
> ...supports websites based on multi-page (MPA) and single page (SPA) architectures, out of the box...

File: docs/supported-technologies.asciidoc [hunk 2]
```diff
@@ -117,6 +117,12 @@ The agent supports <<framework-integrations,integrations with certain frameworks
...
+
+* They group pages by routes better. This results in `transaction.name` mapped to the actual application route path the users have declared in their SPA application.
+* They exactly know when a SPA navigation is rendered, the core RUM agent does not know that and waits for the last network request before considering the SPA transition is finished. The integration packages on the other hand hook into the component lifecycle and measures the duration of the SPA transition correctly.
```
Comment (paulb-elastic) [inline docs/supported-technologies.asciidoc:123]: nit typo, change "...when a SPA navigation..." to "...when an SPA navigation..."

File: docs/supported-technologies.asciidoc [hunk 3]
```diff
@@ -117,6 +117,12 @@ The agent supports <<framework-integrations,integrations with certain frameworks
...
+* They improve error capturing. For instance, when an error is thrown inside an Angular application, the default error handler doesn’t rethrow them as browser events, which makes them invisible to the core RUM agent. The integration automatically captures those errors, too.
+
```
Comment (paulb-elastic) [inline docs/supported-technologies.asciidoc:125]: Maybe add a final line;
> To be clear, the core RUM agent can be used on its own with SPA sites, however also using the framework integration provides the additional capabilities shown.

**Review by paulb-elastic — APPROVED** [review]
Made a few comments, but nothing that blocks a LGTM
Would be good to still get @vigneshshanmugam's review on the detail

File: docs/supported-technologies.asciidoc (hunk 3 above)
Comment (devcorpio) [reply docs/supported-technologies.asciidoc:125]: I applied the three changes 🎉

**Comment by devcorpio** [comment]
Hi @florent-leborgne,

Could you please review this PR?

Thanks,
Alberto

**Comment by devcorpio** [comment]
thank you @florent-leborgne !

File: docs/supported-technologies.asciidoc [hunk 4]
```diff
@@ -117,6 +117,13 @@ The agent supports <<framework-integrations,integrations with certain frameworks
...
+* They group pages by routes better. This results in `transaction.name` mapped to the actual application route path the users have declared in their SPA application.
+* They exactly know when an SPA navigation is rendered, the core RUM agent does not know that and waits for the last network request before considering the SPA transition is finished. The integration packages on the other hand hook into the component lifecycle and measures the duration of the SPA transition correctly.
+* They improve error capturing. For instance, when an error is thrown inside an Angular application, the default error handler doesn’t rethrow them as browser events, which makes them invisible to the core RUM agent. The integration automatically captures those errors, too.
+
+To be clear, the core RUM agent can be used on its own with SPA sites, however also using the framework integration provides the additional capabilities shown.
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc:126]:
```suggestion
The core RUM agent supports websites based on multi-page (MPA) and single-page architectures (SPA). The core RUM agent can be used on its own with SPA sites, but we recommend using our framework-specific integrations for the following reasons:

* **Pages are better grouped by route**: This results in the `transaction.name` field mapping to the actual application route path declared in the SPA application.
* **Exact SPA navigation rendering**: Integration packages hook into the component lifecycle and measure the duration of the SPA transaction correctly. The core RUM agent is unable to determine when an SPA navigation is rendered and instead waits for the last network request before considering the SPA transition is finished.
* **Improved error capturing**: For example, when an error is thrown inside an Angular application, the framework integration will automatically capture it. The core RUM agent, however, is unable to capture the error as the default error handler doesn’t rethrow the error as a browser event.
```

**Review by bmorelli25 — APPROVED** [review]
I'm sorry, this one snuck by my email filters. I have one big recommendation for you. I recommend we rewrite and lead with the benefit of the integration, not the downside of the core agent.

File: docs/supported-technologies.asciidoc (hunk 4 above)
Comment (devcorpio) [reply docs/supported-technologies.asciidoc:126]: thank you for the rewriting @bmorelli25 !

**Comment by devcorpio** [comment]
preview with recommendations applied:

Screenshot 2023-05-04 at 21 10 49

**Pull request merged**